man/sysupdate.d.xml

```diff
@@ -820,7 +825,7 @@ MatchPattern=foobarOS_@v.efi.xz
 
 [Target]
 Type=regular-file
-Path=/efi/EFI/Linux
+Path=$BOOT/EFI/Linux
```

Comment:
hmm, i guess having this does make some sense, even though I don't particularly like it...

not sure about spelling out $BOOT here. elswehere we use `%`-style specifiers, but i don't think this would be right here, since we interpret those usually at config read time, but the relatively expensive $BOOT resolution is probably somethign we shouldn't do a parse time, but when actually opening the path.

Reply:
I find this quite confusing though, it's not immediately obvious and doesn't leave much room for further expansion, isn't it better to just bite the bullet and implement specifiers?

Reply:
I originally didn't go for specifiers either. Partially for Lennart's reasoning, but I had a couple other reasons too:

It would reserve a character for this extremely niche situation. I cannot think of anywhere else a specifier that resolves to $BOOT can be useful. But if I made it a specifier that character would be unusable for any other specifiers anywhere else, no?

Specifiers can go in the middle of a string, which (unlike things like version numbers, architectures, etc) makes no sense here.

Reply:
I am pretty sure that at some point support for other similar cases will be needed

Reply:
> I find this quite confusing though, it's not immediately obvious and doesn't leave much room for further expansion, isn't it better to just bite the bullet and implement specifiers?

You mean specifiers as in `%b`? the problem is that so far we clearly said those are interpreted at parse time, i.e. are relatively dumb. I doubt this is the right choice for the search for ESP stuff, it's just too heavy an op, because it queries udev and shit...

Reply:
note that ^ we don't use so far otherwise, afaics

Another option could be to add BootPartitionPath= i.e. make this a highj-level concept:

Path= means relative to /, and BootPartitionPath= means relative to $BOOT.

Might be safer, dunno?

Reply:
hmm, your idea would probably be more future-proof with an enum, i.e. we could then do:

```
PathRelativeTo=root|boot|esp|xbootldr
```

I think I like this approach best. @bluca what do you think?

Reply:
That sounds much better - give we have partitions types defined in the DPS, what about making it match Repart's [Partition] Type= in terms of what it accepts? And maybe just PartitionType= or so? For consistency across components

Reply:
Not sure I'm following on `PartitionType`:

- `linux-generic`: What path does that map to?
- `*-verity`: We can't put files into there
- `*-verity-sig`: Ditto
- `swap`: Ditto
- Why would someone write this:

```ini
Path=/myservice
PartitionType=srv
```

over this:

```ini
Path=/srv/myservice
```

- How would that interact with sysupdate's existing `Path=auto` mechanism?
- We'd also need to add a special PartitionType for `boot` ($BOOT)

Reply:
By that logic, then there's no need for this at all, and one can just write Path=/efi/EFI/linux or Path=/boot/EFI/linux depending on how their system is configured

Reply:
> depending on how their system is configured

These are the key words... How should a distro know ahead of time which a user's system is using?

The BLS tells me very clearly that I need to put my UKIs in $BOOT, which may be /efi or it may be /boot depending on whether or not the XBOOTLDR partition exists. This is entirely out of my control:

- what if the user is dual-booting and the existing ESP is tiny? 
- what if another distro already created XBOOTLDR?

I'd agree with you if $BOOT was an actual single partition defined by the DPS. But it isn't: it's and alias for one of two other partitions defined by the DPS. If I wanted to always just update the ESP and never XBOOTLDR then yes I'd absolutely just write `/efi/EFI/Linux`

Reply:
either it's configured explicitly, and then just use the configure path as-is, or if it's auto then follow the spec and use xbootldr if there else the esp

Reply:
Sorry, but I'm confused by your response: if _what_ is auto? It seems like you're describing what I've implemented in this PR... So the remaining question is what config file syntax I should implement

Reply:
I think it definitely makes sense to reuse the id strings of the partition types for this new enum, but while there's an overlap between the two there are certainly some values that makes sense on one enum but not the other and vice versa.

i.e. $BOOT as a concept (i.e. that refers to either XBOOTLDR or ESP, depending on what's available) doesn't make sense for the partition table. Moreover, many of the partition types don't really make sense as file system paths (e.g. the verity ones or swap). 

Hence, I'd just start out with a simple enum here, with only the values we care about, and a comment that makes clear we should follow the same naming as the partition type list fro gpt.c, i.e.

```c
typedef enum PathRelativeTo {
        /* Please make sure to folow the naming of the corresponding PartitionDesignator enum values, where this makes sense, like for the following three. */
        PATH_RELATIVE_TO_ROOT,
        PATH_RELATIVE_TO_ESP,
        PATH_RELATIVE_TO_XBOOTLDR,
        PATH_RELATIVE_TO_BOOT, /* This one refers to either XBOOTLDR or ESP depending what is available. It has no direct counterpart in PartitionDesignator */
        _PATH_RELATIVE_TO_MAX,
        _PATH_RELATIVE_TO_INVALID = -1,
} PathRelativeTo;
```

And if you feel like it you could even add a test that the names match somewhere:

```c
assert(streq_ptr(path_relative_to_to_string(PATH_RELATIVE_TO_ROOT), partition_designator_to_string(PARTITION_ROOT));
assert(streq_ptr(path_relative_to_to_string(PATH_RELATIVE_TO_ESP), partition_designator_to_string(PARTITION_ESP));
assert(streq_ptr(path_relative_to_to_string(PATH_RELATIVE_TO_BOOT), partition_designator_to_string(PARTITION_BOOT));
```

just to make sure this doesn't diverge.

But still, i'd keep the enums separate, since they only have a limited intersction, and both have entries the other doesn't have.